### PR TITLE
Infrastructure for the SpecViz Statistics Widget

### PR DESCRIPTION
**Pull request by robelgeda**
This PR introduces infrastructure for the SpecViz widget.

The Stat Widget displaying stats under the selected region:
screen shot 2018-08-23 at 12 56 27 pm

Tooltips describing functions used to compute stats:
 
screen shot 2018-08-23 at 12 56 48 pm

Error messages in status bar:
screen shot 2018-08-23 at 12 53 40 pm

### CODE REVIEW

File: specviz/widgets/workspace.py [hunk 1]
```diff
@@ -87,12 +129,27 @@ def current_item(self):
...
+        to be made before the local stat_widget is set.
+        """
+        if self.stat_widget is not None:
+            self.stat_widget.update_statistics()
+
+    def _set_list_view_model(self, model):
+        self.list_view.setModel(model)
+
+        # Set up connections
+        self.list_view.selectionModel().currentChanged.connect(self.update_stats)
```
Comment (robelgeda) [inline specviz/widgets/workspace.py]: signal connection

File: specviz/widgets/workspace.py [hunk 2]
```diff
@@ -105,6 +162,9 @@ def add_plot_window(self):
         # Subscribe this new plot window to list view item selection events
         self.list_view.selectionModel().currentChanged.connect(plot_window._on_current_item_changed)
 
+        plot_window.plot_widget.roi_moved.connect(self.update_stats)
```
Comment (robelgeda) [inline specviz/widgets/workspace.py]: signal connection

File: specviz/widgets/workspace.py [hunk 3]
```diff
@@ -40,12 +40,21 @@ def __init__(self, *args, **kwargs):
...
+        self._statistics = None
+
         # Don't expand mdiarea tabs
         self.mdi_area.findChild(QTabBar).setExpanding(True)
 
         # When the current subwindow changes, mount that subwindow's proxy model
         self.mdi_area.subWindowActivated.connect(self._on_sub_window_activated)
 
+        # When the current subwindow changes, update the stat widget
+        self.mdi_area.subWindowActivated.connect(self.update_stats)
```
Comment (robelgeda) [inline specviz/widgets/workspace.py]: signal connections

File: specviz/widgets/workspace.py [hunk 4]
```diff
@@ -40,12 +40,21 @@ def __init__(self, *args, **kwargs):
...
+
+        # When current item changes, update the stat widget
+        self.current_item_changed.connect(self.update_stats)
```
[the remark this answers is not in the document]
Comment (nmearl) [reply specviz/widgets/workspace.py]: e.g.

```python
workspace.current_item_changed.connect(self.update_stats)
```

[the remark this answers is not in the document]
Comment (robelgeda) [reply specviz/widgets/workspace.py]: @nmearl  I have to think about this one. The problem is that the connections need to happen within functions in the `worspace` (for example look at `add_plot_window`). I will try to find a clever way to resolve this.

**Comment by robelgeda** [comment]
@nmearl  I made it as plugin agnostic as possible by adding two signals:
```
list_view_model_changed = Signal()
plot_window_added = Signal(PlotWindow)
```

**Pull request merged**